flink-streaming-java/src/test/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGeneratorTest.java

```diff
@@ -1201,6 +1243,33 @@ private void verifyFractions(
                 delta);
     }
 
+    @Test
+    public void testSetNonDefaultSlotSharingInHybridMode() {
+        Configuration configuration = new Configuration();
+        // set all edge to HYBRID result partition type.
+        configuration.set(
+                ExecutionOptions.BATCH_SHUFFLE_MODE,
+                BatchShuffleMode.EXPERIMENTAL_ALL_EXCHANGES_HYBRID);
+
+        final StreamGraph streamGraph = createStreamGraphForSlotSharingTest(configuration);
+        // specify slot sharing group for map1
+        streamGraph.getStreamNodes().stream()
+                .filter(n -> "map1".equals(n.getOperatorName()))
+                .findFirst()
+                .get()
+                .setSlotSharingGroup("testSlotSharingGroup");
+
+        try {
+            StreamingJobGraphGenerator.createJobGraph(streamGraph);
+            fail("hybrid shuffle mode with non default slot sharing group should failed.");
+        } catch (IllegalStateException e) {
+            assertTrue(
+                    e.getMessage()
+                            .contains(
+                                    "hybrid shuffle mode currently does not support setting slot sharing group"));
```

Comment:
I'd suggest to not verify the complete error message.

Reply:
We can use the annotation `@Test(expected = IllegalStateException.class)`. In this way, we don't need the `try-catch` and `fail()`.

Reply:
This should use assertj's assertThatThrownby()

Reply:
> This should use assertj's assertThatThrownby()

nice suggestion, i have fixed it.